src/settings.c

```diff
@@ -384,11 +353,10 @@ static void settings_write_callback(uint16_t sender_id, uint8_t len, uint8_t *ms
   /* Extract parameters from message:
    * 3 null terminated strings: section, setting and value
    */
-  const char *section = NULL;
-  const char *name = NULL;
-  const char *value = NULL;
-  if (settings_parse((char *)msg, len, &section, &name, &value) != 0) {
-    ctx->api_impl.log(log_warning, "settings write message failed");
+  const char *section = NULL, *name = NULL, *value = NULL, *type = NULL;
+  int res = settings_parse((char *)msg, len, &section, &name, &value, &type);
+  if (res < 3) {
```

Comment:
I'm not sure what i think of `res < SETTINGS_TOKENS_VALUE`. Does it make sense for `res` to be "less than" `VALUE`? (In this case, implying that `VALUE` was missing)
Just a possibility

Reply:
Not sure if I understood the review comment fully but I added further commenting and unified the styles for the two occurrences where `settings_parse` is called.